src/ostorlab/scanner/callbacks.py

```diff
@@ -1,6 +1,160 @@
 """Defines call back to trigger a scan after receiving a startAgentScan messages in the NATS."""
+import logging
+import ipaddress
+from typing import List
 
+from ostorlab.runtimes import registry
+from ostorlab.runtimes import definitions
+from ostorlab.runtimes import runtime
+from ostorlab.cli import install_agent
+from ostorlab.assets import asset
+from ostorlab.assets import ipv4
+from ostorlab.assets import ipv6
+from ostorlab.assets import android_store
+from ostorlab.assets import android_aab
+from ostorlab.assets import android_apk
+from ostorlab.assets import file
+from ostorlab.assets import ios_ipa
+from ostorlab.assets import domain_name
+from ostorlab.assets import link as link_asset
+from ostorlab.assets import ios_store
+from ostorlab.assets import agent as agent_asset
 
-async def start_scan():
-    """TODO(deadly-panda): Add the missing start scan logic."""
-    pass
+
+logger = logging.getLogger(__name__)
+
+
+def _install_agents(runtime_instance: runtime.Runtime, agents) -> None:
+    """Trigger installation of the agents that will run the scan."""
+    try:
+        runtime_instance.install()
+        for agent in agents:
+            install_agent.install(agent.key, agent.version)
+    except install_agent.AgentDetailsNotFound:
+        logger.warning("agent %s not found on the store", agent.key)
+
+
+def _prepare_ip_asset(ip_request) -> asset.Asset:
+    ip_network = ipaddress.ip_network(ip_request.host, strict=False)
+    if ip_network.version == 4:
+        return ipv4.IPv4(
+            host=ip_network.network_address.exploded,
+            mask=ip_request.mask or str(ip_network.prefixlen),
+        )
+    elif ip_network.version == 6:
+        return ipv6.IPv6(
+            host=ip_network.network_address.exploded,
+            mask=ip_request.mask or str(ip_network.prefixlen),
+        )
+    else:
+        raise ValueError(f"Invalid Ip address {ip_request.host}")
+
+
+def _asset_instances_from_request(request) -> List[asset.Asset]:
+    assets = []
+    asset_type = request.WhichOneof("asset")
+    if asset_type in ("ip", "ip4v", "ipv6"):
```

Comment:
Can you please use pattern matching for improved readability.

Reply:
Are we dropping support for py3.9 any time soon?